modules/objdetect/src/aruco/aruco_detector.cpp

```diff
@@ -1000,7 +1000,7 @@ static inline void _projectUndetectedMarkers(const Board &board, InputOutputArra
                                              OutputArray undetectedMarkersIds) {
     Mat rvec, tvec; // first estimate board pose with the current avaible markers
     Mat objPoints, imgPoints; // object and image points for the solvePnP function
-    board.matchImagePoints(detectedCorners, detectedIds, objPoints, imgPoints);
```

Comment:
Why does it change the function behaviour?

Reply:
The issue #24127 uses `CharucoBoard` and ChArUco board call `matchImagePoints` for charuco corners, but to refine ArUco markers we need to call `matchImagePoints` for ArUco corners. The method `matchImagePoints` implemented in Pimpl and we need to create `Board` object.

`refineDetectedMarkers` must always call `matchImagePoints` from the base class `Board`.

Reply:
It should be added to the implementation as comment.

Reply:
@AleksandrPanov Please add this information to the code as comment. Current comment is not informative.